diff.py

```diff
@@ -556,10 +572,14 @@ def table(
 class AnsiFormatter(Formatter):
     column_width: int
 
-    def apply(self, line: str, f: Format) -> str:
-        if f == Format.NONE:
-            return line
-        return f"{f.ansi_code}{line}{Style.RESET_ALL}"
+    def apply(self, text: Text) -> str:
```

Comment:
maybe cleaner to have this receive `chunk` and `f` and put the loop in the caller